autoemulate/experimental/data/utils.py

```diff
@@ -183,12 +183,333 @@ def _denormalize(
     ) -> TensorLike:
         return (x * x_std) + x_mean
 
-    # TODO: consider possible method for predict
-    # def convert_x(self, y: np.ndarray | torch.Tensor | Data) -> torch.Tensor:
-    #     if isinstance(y, np.ndarray):
-    #         y = torch.tensor(y, dtype=torch.float32)
-    #     else:
+
+class ValidationMixin:
+    """
+    Mixin class for validation methods.
+    This class provides static methods for checking the types and shapes of
+    input and output data, as well as validating specific tensor shapes.
+    """
+
+    @staticmethod
+    def _check(x: TensorLike, y: TensorLike | None):
+        """
+        Check the types and shape are correct for the input data.
+        Checks are equivalent to sklearn's check_array.
```

Comment:
The below code attempts to do what sklearn was doing with `check_array`, but considering that we have tensors as oppose to numpy arrays - I'm open to people suggesting this isn't needed or should be changed

Reply:
I think this looks great and captures what we expect for x and y.